### PR TITLE
Use RawHttpMessageHandler and VssHttpRetryMessageHandler in ResultsHttpClient

### PR DESCRIPTION
**Pull request by yacaovsnc**
Consolidate http client implementation in `ResultsHttpClient`.  Use the standard message handlers in Results as well.

### CODE REVIEW

File: src/Sdk/WebApi/WebApi/RawHttpClientBase.cs [hunk 1]
```diff
@@ -68,6 +68,20 @@ public class RawHttpClientBase : IDisposable
             m_formatter = new VssJsonMediaTypeFormatter();
         }
 
+        protected RawHttpClientBase(
+            Uri baseUrl,
+            HttpMessageHandler pipeline,
+            int timeoutSeconds,
+            bool disposeHandler)
+        {
+            m_client = new HttpClient(pipeline, disposeHandler);
```
Comment (TingluoHuang) [inline src/Sdk/WebApi/WebApi/RawHttpClientBase.cs:77]: i would suggest we use the same message handler, and not the raw httpclient

File: src/Sdk/Common/Common/Authentication/NoOpCredentials.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GitHub.Services.Common
+{
+    // Set of classes used to bypass token operations
+    // Results Service and External services follow a different auth model but
+    // we are required to pass in a credentials object to create a RawHttpMessageHandler
+    public class NoOpCredentials : FederatedCredential
```
Comment (TingluoHuang) [inline src/Sdk/Common/Common/Authentication/NoOpCredentials.cs:10]: do you need to have `FedCred` or just use `IssuedTokenCredential`?

Comment (TingluoHuang) [reply src/Sdk/Common/Common/Authentication/NoOpCredentials.cs:10]: then you can always return false for `IsAuthenticationChallenge()`

File: src/Runner.Sdk/Util/VssUtil.cs [hunk 3]
```diff
@@ -85,6 +85,35 @@ public static void InitializeVssClientSettings(List<ProductInfoHeaderValue> addi
             VssCredentials credentials,
             IEnumerable<DelegatingHandler> additionalDelegatingHandler = null,
             TimeSpan? timeout = null)
+        {
```
Comment (TingluoHuang) [inline src/Runner.Sdk/Util/VssUtil.cs:88]: is these change just moving content around?  create the new `GetHttpRequestSettings` for re-use

File: src/Sdk/Common/Common/Authentication/NoOpCredentials.cs (hunk 2 above)
Comment (yacaovsnc) [reply src/Sdk/Common/Common/Authentication/NoOpCredentials.cs:10]: FedCred, it's on the base constructor.  

```
        public RawHttpMessageHandler(
            FederatedCredential credentials,
            RawClientHttpRequestSettings settings,
            HttpMessageHandler innerHandler)
```

File: src/Runner.Common/ResultsServer.cs [hunk 4]
```diff
@@ -52,8 +53,8 @@ public sealed class ResultServer : RunnerService, IResultsServer
 
         public void InitializeResultsClient(Uri uri, string liveConsoleFeedUrl, string token)
         {
-            var httpMessageHandler = HostContext.CreateHttpClientHandler();
-            this._resultsClient = new ResultsHttpClient(uri, httpMessageHandler, token, disposeHandler: true);
+            this._resultsClient = CreateHttpClient(uri, token);
```
Comment (TingluoHuang) [inline src/Runner.Common/ResultsServer.cs:56]: should we have 2 clients for result server, one with Token, one with the NoOpsCredential?

File: src/Runner.Sdk/Util/VssUtil.cs (hunk 3 above)
Comment (yacaovsnc) [reply src/Runner.Sdk/Util/VssUtil.cs:88]: Yup, line 116 is the new method and it's reused in the `ResultsServer`.

**Pull request merged**